conbench/entities/run.py

```diff
@@ -80,57 +86,79 @@ def create(data):
         run.save()
         return run
 
-    def get_baseline_run(self):
+    def get_baseline_run(self) -> Optional["Run"]:
+        """Return the closest ancestor of this Run that:
+
+        - is in this Run's commit ancestry
+        - shares this Run's reason (but see note)
+        - has *any* BenchmarkResults that share the hardware/case/context of *any* of
+            this Run's BenchmarkResults
+
+        Note: if there are no matches for those criteria, we search for ANY run_reason
+        on the default branch. This helps runs on the first commits of pull requests.
+
+        Returns None if there are no matches.
+        Returns a random Run if there are multiple matches.
```

Comment:
er...this seems worrying. I guess this is the case where there are multiple runs on a machine for one commit? In that case, it may be worth prioritizing runs where nothing failed, given one reason for that situation would be something external caused a run to fail, so it got rerun. Though if we're not actually doing that there's less reason to filter, I guess. A warning log or something seems prudent, though.

Reply:
hm, yeah it would have to be 2 runs with the same commit, same reason, same hardware, sharing at least 1 case/context (same as one on the given Run). Oh, I could tiebreak with latest Run timestamp/finished_timestamp?

Reply:
Yeah, I was thinking about timestamps, but unfortunately it seems like the two cases I can think of for this happening suggest opposite heuristics: 

- if something broke and another build was necessary, later is better
- if someone is kicking off builds by hand in a way they shouldn't, earlier is better

Regardless, deterministic results are good, so maybe any choice is better than none

Reply:
>  deterministic results are good

I agree that there should be one definite result here.

In the longer run, maybe all matches should be used in the information flow. What, fundamentally, is a good reason to pick just one data point if all data points are 'good' (meaningful) ones?

Reply:
Cool, I went with tiebreaking by choosing the latest Run.timestamp.

>In the longer run, maybe all matches should be used in the information flow.

Yeah. Right now this powers the `compare ->` button at the top right of a page like [this](https://conbench.ursa.dev/runs/0bc0e39524c14349bbb0556e63799e13/). That compares a run to exactly one other run. But with #384 coming up, in the future there could be a way in the webapp to pick the baseline run out of a list of "recommended" baseline runs. I think we could accommodate this edge case there.